### ISSUE TITLE
Replay generation fails due to missing temporary json file

### ISSUE THREAD

**Issue by marcos-acosta** [awslabs/aws-media-replay-engine#10]
Hello! I created a Replay Request in catchup mode (if that's relevant), but when a new chunk is detected and the replay generation state machine is triggered, it fails because it cannot find a `.json` file in `tmp`. I'm not sure how to debug this, would you be able to provide some insight? Thank you!

Output from `CreateReplay` step:
```
{
  "errorMessage": "[Errno 2] No such file or directory: '/tmp/09744a42-f2ea-42b6-a6d0-c5444dda7f42/Seg_407.7_422.667_1.json'",
  "errorType": "FileNotFoundError",
  "stackTrace": [
    "  File \"/opt/python/aws_lambda_powertools/logging/logger.py\", line 402, in decorate\n    return lambda_handler(event, context, *args, **kwargs)\n",
    "  File \"/var/task/replay_lambda.py\", line 68, in CreateReplay\n    replay_result = replay._create_replay()\n",
    "  File \"/var/task/shared/ReplayEngine.py\", line 607, in _create_replay\n    if not self.initialize_segment_features():\n",
    "  File \"/var/task/shared/ReplayEngine.py\", line 366, in initialize_segment_features\n    if not self.__does_current_segment_have_any_replay_feature(current_segment_cache_file_name):\n",
    "  File \"/var/task/shared/ReplayEngine.py\", line 109, in __does_current_segment_have_any_replay_feature\n    segment_feature_cache_file = open(\n"
  ]
}
```

Logs (csv):
[log-events-viewer-result.csv](https://github.com/awslabs/aws-media-replay-engine/files/11314900/log-events-viewer-result.csv)

**Comment by aravindkaws** [comment]
This error is due to the fact that as and when a segment is being created, the Segment data gets stored in a json file within s3. We refer it to as a cache. Looking at the logs I see the following message which indicates that no cache files have been generated for any of the segments.

Number of cached files Synced into tmp/09744a42-f2ea-42b6-a6d0-c5444dda7f42 = 0

1. Can you confirm that there exists a event bridge rule with the name prefix aws-mre-segment-caching-MRESegmentEndRule and  that the Target Lambda function is [aws-mre-segment-caching-MreSegmentCaching] ? What is the event pattern set for this rule ? 
2. Look into the Logs of the aws-mre-segment-caching-MreSegmentCaching. Do you see errors ?

**Comment by aravindkaws** [comment]
Hi Marcos,

Thats interesting. This feature was introduced in version 2.4.0 and should have been deployed with MRE. What version of MRE are you using ? Meanwhile, for you to deploy this EventBridge rule, the simplest way is to deploy the Caching stack which can be found at aws-media-replay-engine/source/backend/caching. In this folder, just create a Virtual Env, pip install requirements.txt and run the cdk deploy command in the Infrastructure folder.

**Comment by araviraj-aws** [comment]
Thanks @sharashutosh ! In this case, it looks like spaces in the event name could indeed be causing the missing cache file issue (based on the log file uploaded). We'll look into adding a fix for this in the segment caching logic.

And, @marcos-acosta , one possible reason why you couldn't find the `aws-mre-segment-caching-MRESegmentEndRule*` rule in EventBridge is probably because this rule is not present under the `default` event bus but instead under `aws-mre-event-bus`. So, I don't think anything is wrong with your installation of MRE and it's just the spaces in the event name `Random Score Change Event` not being properly handled by the segment caching logic.

**Comment by aravindkaws** [comment]
This issue has been fixed in the MRE release 2.8.0
